[export] Initial draft export

angelayi opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #139383

Differential Revision: [D65288590](https://our.internmc.facebook.com/intern/diff/D65288590)

angelayi committed:
[export] Initial draft export

[ghstack-poisoned]

angelayi commented:
@angelayi has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D65288590).

zou3519 commented on test/export/test_draft_export.py:
```diff
@@ -0,0 +1,229 @@
+            def foo_impl(a, b):
+                return a + b
+
+            @torch.library.register_fake("mylib::foo1")
```
needs lib=lib

zou3519 commented on test/export/test_draft_export.py:
```diff
@@ -0,0 +1,229 @@
+        )
+
+    def test_missing_meta_kernel(self):
+        with torch.library._scoped_library("mylib", "FRAGMENT") as lib:
```
can you test something with torch.library.custom_op? 

Also in general we should try to use torch.library.custom_op, unless we're testing the other torch.library APIs explicitly

zou3519 commented on torch/export/_draft_export.py:
```diff
@@ -0,0 +1,285 @@
+            return f"""Missing fake kernel.
+    torch.ops.{op} is missing a fake kernel implementation.
+
+    Here's a template for registering a fake kernel implementation. Please refer to https://docs.google.com/document/d/1_W62p8WJOQQUzPsJYa7s701JXt0qf2OfLub2sbkHOaU/edit#heading=h.ahugy69p2jmz for more detailed instructions.
```
" Here's a template for registering a fake kernel implementation" -> where is the template?

angelayi replied:
oops I used to have it with the previous op profiles. I'll add it back later since we don't have this info yet.

zou3519 commented on torch/export/_draft_export.py:
```diff
@@ -0,0 +1,285 @@
+        report = DraftExportReport(failures, str_to_filename)
+
+    ep._report = report
+    print(report)
```
do we intend for this to always print?

angelayi replied:
I was thinking yes, since users should see the errors... Maybe I could warning instead.

zou3519 reviewed (approved):
looks reasonable

angelayi committed:
Update on "[export] Initial draft export"


Differential Revision: [D65288590](https://our.internmc.facebook.com/intern/diff/D65288590)

[ghstack-poisoned]

angelayi commented:
@angelayi has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D65288590).

angelayi commented:
@pytorchbot merge
